src/python/pants/build_graph/address.py

```diff
@@ -244,18 +244,20 @@ class BuildFileAddress(Address):
   :API: public
   """
 
-  def __init__(self, build_file, target_name=None):
+  def __init__(self, build_file=None, rel_path=None, target_name=None):
```

Comment:
since this is `:API: public`, you should probably move the new `rel_path` kwarg to the end so as not to break any existing positional usage.